### PR TITLE
Fixing default semantic use case integration test

### PR DESCRIPTION
**Pull request by joshpalis**
### Description
Addresses failed distribution build test : https://github.com/opensearch-project/flow-framework/issues/599#issuecomment-2014020214

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/flowframework/FlowFrameworkRestTestCase.java [hunk 1]
```diff
@@ -706,4 +708,22 @@ protected GetPipelineResponse getPipelines() throws IOException {
...
+            "",
+            ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, ""))
+        );
+        List<Object> pluginsList = JsonXContent.jsonXContent.createParser(
+            NamedXContentRegistry.EMPTY,
+            LoggingDeprecationHandler.INSTANCE,
+            response.getEntity().getContent()
+        ).list();
+        return pluginsList.stream().map(o -> ((Map<String, Object>) o).get("component").toString()).collect(Collectors.toList());
+    }
```
Comment (owaiskazi19) [inline src/test/java/org/opensearch/flowframework/FlowFrameworkRestTestCase.java:728]: We can use the same method [here](https://github.com/opensearch-project/flow-framework/blob/main/src/test/java/org/opensearch/flowframework/FlowFrameworkPluginIT.java#L23) as well

**Comment by owaiskazi19** [comment]
One question. How the test is failing for distribution build though? Since it contains all the plugins already.

**Comment by joshpalis** [comment]
> One question. How the test is failing for distribution build though? Since it contains all the plugins already.

So the distribution build is failing since all plugin integration tests run on the same cluster, this particular integration test of ours assumes the neural search plugin is not in our cluster since our we dont have that plugin dependency in our build.gradle.

**Comment by owaiskazi19** [comment]
> So the distribution build is failing since all plugin integration tests run on the same cluster, this particular integration test of ours assumes the neural search plugin is not in our cluster since our we dont have that plugin dependency in our build.gradle.

How did this test pass on GHA then?

**Comment by joshpalis** [comment]
> How did this test pass on GHA then?

When we run integration tests on our CI, the cluster only has ML Commons and Flow Framework

**Comment by owaiskazi19** [comment]
> When we run integration tests on our CI, the cluster only has ML Commons and Flow Framework

I see it was a failure test case that's why no issue on our GHA

**Review by owaiskazi19 — APPROVED** [review]

**Review by amitgalitz — APPROVED** [review]

**Comment by joshpalis** [comment]
will raise manual PRs for the backports since this modifies the FlowFrameworkRestTestCase file

**Pull request merged**